src/app.ts

```diff
@@ -498,38 +473,83 @@ export class App {
     parameters: Record<string, JsonSchema> = {},
     schemas: Record<string, JsonSchema>
   ) {
-    _.forEach(resources, (resource, resourceName) => {
-      const resourceInterfaceName = getResourceTypeName(resourceName);
+    // Accumulates the request method parameter types, to be written after all of the resource interfaces.
+    const requestTypeWriters: (() => void)[] = [];
 
+    // Write all resource interfaces
+    _.forEach(resources, (resource, resourceName) => {
       if (resource.resources) {
         this.writeResources(out, resource.resources, parameters, schemas);
       }
 
-      out.interface(resourceInterfaceName, () => {
+      out.interface(getResourceTypeName(resourceName), () => {
+        if (resource.resources) {
+          _.forEach(resource.resources, (_, childResourceName) => {
+            const childResourceInterfaceName = getResourceTypeName(
+              childResourceName
+            );
+            out.property(childResourceName, childResourceInterfaceName);
+          });
+        }
+
         _.forEach(resource.methods, method => {
           if (method.description) {
             out.comment(formatComment(method.description));
           }
 
-          const requestRef = method.request?.$ref;
-          const requestParameters: Record<string, JsonSchema> = sortObject({
-            ...parameters,
-            ...method.parameters,
+          // Construct request type, e.g. get(...) on InvitationsResource -> GetInvitationsRequest
+          const methodType = checkExists(getName(method.id));
+          const methodParts = method.id?.split('.');
```

Comment:
As raised in the previous PR, we weren't including the parent path down to the resource method, which meant we'd get some interface duplication, e.g. `NotesResource` in the containeranalysis API. 

Using `method.id` (looks something like `containeranalysis.projects.notes.occurrences.list`) instead of just `resourceName` (just `notes`) will let us process it into something like `ListProjectsNotesOccurencesRequest`, which is fully namespaced and shouldn't conflict.